rasa/core/test.py

```diff
@@ -245,23 +257,21 @@ def _clean_entity_results(
 
 def _collect_user_uttered_predictions(
     event: UserUttered,
+    predicted: Dict[Text, Any],
     partial_tracker: DialogueStateTracker,
     fail_on_prediction_errors: bool,
 ) -> EvaluationStore:
     user_uttered_eval_store = EvaluationStore()
 
-    intent_gold = event.parse_data.get("true_intent")
-    predicted_intent = event.parse_data.get("intent", {}).get("name")
-
-    if not predicted_intent:
-        predicted_intent = [None]
+    intent_gold = event.intent.get("name")
```

Comment:
where was the `"true_intent"` originally coming from and why did we switch?

Reply:
the true intent was attached in the training data reader, it wasn't necessary (as the value was always set to the same value as the intent, which makes sense when reading the data). not sure who implemented it, but it was a workaround - data gets pushed through the evaluation slightly different separating loaded data from predictions.